tests/test_marquez_dag.py

```diff
@@ -66,55 +65,54 @@ def clear_db_airflow_dags(session=None):
     session.query(TaskInstance).delete()
 
 
-class DummyExtractor(BaseExtractor):
-    operator_class = DummyOperator
-
-    def __init__(self, operator):
-        super().__init__(operator)
-
-    def extract(self) -> [StepMetadata]:
-        return [StepMetadata(
-            name=get_job_name(task=self.operator),
-            context={
-                "extract": "extract"
-            }
-        )]
-
-    def extract_on_complete(self, task_instance) -> [StepMetadata]:
-        return [StepMetadata(
-            name=get_job_name(task=self.operator),
-            context={
-                "extract_on_complete": "extract_on_complete"
-            }
-        )]
+@provide_session
+def test_new_run_id(clear_db_airflow_dags, session=None):
+    dag = DAG(
+        DAG_ID,
+        schedule_interval='@daily',
+        default_args=DAG_DEFAULT_ARGS,
+        description=DAG_DESCRIPTION
+    )
+    run_id = dag.new_run_id()
+    assert UUID(run_id).version == 4
 
 
+# tests a simple workflow with default extraction mechanism
+@mock.patch('marquez_airflow.DAG.new_run_id')
 @mock.patch('marquez_airflow.DAG.get_or_create_marquez_client')
 @provide_session
-def test_marquez_dag(mock_get_or_create_marquez_client,
+def test_marquez_dag(mock_get_or_create_marquez_client, mock_uuid,
                      clear_db_airflow_dags, session=None):
+
+    dag = DAG(
+        DAG_ID,
+        schedule_interval='@daily',
+        default_args=DAG_DEFAULT_ARGS,
+        description=DAG_DESCRIPTION
+    )
     # (1) Mock the marquez client method calls
     mock_marquez_client = mock.Mock()
     mock_get_or_create_marquez_client.return_value = mock_marquez_client
+    run_id_completed = "my-test_marquez_dag-uuid-completed"
```

Comment:
_Minor:_ Should we make the test run IDs valid uuids to align with the contract of the Marquez API?

Reply:
since the implementation does not depend on it I think this is more readable here.
The other test verifies that we actually generate type4 uuids

Reply:
Sounds good 👍